### PR TITLE
[Focus Mode] New images and content for JS reference article

### PR DESCRIPTION
**Pull request by captainbrosset**
This PR contains all new images (and some new content) to make the JS reference article match the new Focus Mode UI.

* Before: https://learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/javascript/reference
* After: https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/javascript/reference?branch=pr-en-us-2894

AB#47262213

### CODE REVIEW

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 1]
```diff
@@ -62,67 +70,77 @@ Use a line-of-code breakpoint when you know the exact region of code that you ne
...
 ```
 
-You are paused on `A`.  After you click **Step over**, DevTools runs all the code in the function that you are stepping over, which is `B` and `C`.  DevTools then pauses on `D`.
+You are paused on `A`.  After you click **Step over**, DevTools runs all the code in the `getNumber1()` function and then pauses on `B`. If you click **Step over** again, DevTools run all the code in the `getNumber2()` function and then pauses on `C`.
 
 
 <!-- ------------------------------ -->
 #### Step into line of code
 
 When paused on a line of code containing a function call that is related to the problem you are debugging, click the **Step into** (![Step into](./reference-images/step-into-icon.png)) button to investigate that function further.
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:126]:
```suggestion
When paused on a line of code containing a function call that is related to the problem you are debugging, click the **Step into** (![Step into](./reference-images/step-into-icon.png)) button to investigate that function further:
```

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 2]
```diff
@@ -62,67 +70,77 @@ Use a line-of-code breakpoint when you know the exact region of code that you ne
...
+    return inputs[1].value;
 }
 ```
 
-You are paused on `A`.  After you click **Step over**, DevTools runs all the code in the function that you are stepping over, which is `B` and `C`.  DevTools then pauses on `D`.
+You are paused on `A`.  After you click **Step over**, DevTools runs all the code in the `getNumber1()` function and then pauses on `B`. If you click **Step over** again, DevTools run all the code in the `getNumber2()` function and then pauses on `C`.
 
 
 <!-- ------------------------------ -->
 #### Step into line of code
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:124]:
```suggestion
#### Step into a line of code
```

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 3]
```diff
@@ -62,67 +70,77 @@ Use a line-of-code breakpoint when you know the exact region of code that you ne
...
 
 <!-- ====================================================================== -->
 ## Step through code
 
-After your code is paused at a breakpoint, step through the code, one line at a time, investigating control flow and property values along the way.
+After your code is paused at a breakpoint, step through the code, one line at a time, investigating control flow and property values along the way. To learn how to pause at a breakpoint, see [Set a breakpoint, to pause code](#set-a-breakpoint-to-pause-code), above.
 
 
 <!-- ------------------------------ -->
 #### Step over line of code
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:97]:
```suggestion
#### Step over a line of code
```

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 4]
```diff
@@ -134,37 +152,45 @@ You are paused on `A`.  After you click **Step into**, DevTools runs this line o
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:152]: 151 change heading from:
Step out of line of code
to:
Step out of a line of code
or:
Step out from a line of code

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 5]
```diff
@@ -175,42 +201,41 @@ To pause on the first line of the top function in your call stack, while paused
...
 
-You are paused on `A`.  After selecting **Restart frame**, you should be paused on `B`, without ever setting a breakpoint or choosing **Resume script execution**.
+You are paused on `A`.  After selecting **Restart frame**, you're be paused on `B`, without having to set a breakpoint on that line and choosing **Resume script execution**.
 
-![Selecting Restart frame](./reference-images/javascript-source-page-debugger-restart-frame.png)
+![Selecting Restart frame](./reference-images/restart-frame.png)
 
 
 <!-- ------------------------------ -->
 #### Resume script runtime
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:218]: this heading is meaningless to me.  nonstandard wording?  can we express in relatable terms; what would classroom instructor say to students here?

Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:218]:
```suggestion
#### Resume script execution
```

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 6]
```diff
@@ -264,14 +287,14 @@ This should be moved to an "Export debug data" H2 section when there is enough c
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:287]: change 259 from
Copy stack trace
to
Copy a stack trace

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 7]
```diff
@@ -175,42 +201,41 @@ To pause on the first line of the top function in your call stack, while paused
...
 #### Force script runtime
 
 To ignore all breakpoints and force your script to continue to run, click and hold the **Resume script execution** (![Resume script execution](./reference-images/resume-script-run-icon.png)) button and then click the **Force script execution** (![Force script execution](./reference-images/force-script-run-icon.png)) button.
 
-![Clicking the Force script execution button](./reference-images/javascript-sources-get-started-js-force-script-runtime.png)
+![Clicking the Force script execution button](./reference-images/force-script-runtime.png)
 
 
 <!-- ------------------------------ -->
 #### Change thread context
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:234]:
```suggestion
#### Change the thread context
```

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 8]
```diff
@@ -175,42 +201,41 @@ To pause on the first line of the top function in your call stack, while paused
...
 #### Resume script runtime
 
 To continue the runtime after a pause of your script, click the **Resume script execution** (![Resume script execution](./reference-images/resume-script-run-icon.png)) button.  DevTools runs the script up until the next breakpoint, if any.
 
-![Clicking the Resume script execution button](./reference-images/javascript-sources-get-started-js-resume-script-runtime.png)
+![Clicking the Resume script execution button](./reference-images/resume-script-runtime.png)
 
 
 <!-- ------------------------------ -->
 #### Force script runtime
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:226]:
```suggestion
#### Force script execution
```

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 9]
```diff
@@ -175,42 +201,41 @@ To pause on the first line of the top function in your call stack, while paused
...
+![Clicking the Force script execution button](./reference-images/force-script-runtime.png)
 
 
 <!-- ------------------------------ -->
 #### Change thread context
 
 When working with web workers or service workers, click on a context listed in the **Threads** pane to switch to that context.  The blue arrow icon represents which context is currently selected.
 
-![The Threads pane](./reference-images/javascript-sources-main-min-js-threads.png)
+![The Threads pane](./reference-images/threads.png)
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:238]: add red box on blue arrow

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 10]
```diff
@@ -175,42 +201,41 @@ To pause on the first line of the top function in your call stack, while paused
...
 To ignore all breakpoints and force your script to continue to run, click and hold the **Resume script execution** (![Resume script execution](./reference-images/resume-script-run-icon.png)) button and then click the **Force script execution** (![Force script execution](./reference-images/force-script-run-icon.png)) button.
 
-![Clicking the Force script execution button](./reference-images/javascript-sources-get-started-js-force-script-runtime.png)
+![Clicking the Force script execution button](./reference-images/force-script-runtime.png)
 
 
 <!-- ------------------------------ -->
 #### Change thread context
 
 When working with web workers or service workers, click on a context listed in the **Threads** pane to switch to that context.  The blue arrow icon represents which context is currently selected.
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:236]:
```suggestion
When working with web workers or service workers, click on a context listed in the **Threads** pane to switch to that context.  The blue arrow icon represents which context is currently selected:
```

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 11]
```diff
@@ -223,33 +248,31 @@ While paused on a line of code, use the **Scope** pane to view and edit the valu
...
 ## View the call stack
 
 While paused on a line of code, use the **Call Stack** pane to view the call stack that got you to this point.
 
-<!--If you are working with async code, check the **Async** checkbox to enable async call stacks.  -->
-
 Click an entry to jump to the line of code where that function was called.  The blue arrow icon represents which function DevTools is currently highlighting.
 
-![The Call Stack pane](./reference-images/javascript-glitch-debug-js-sources-get-started-inputs-are-empty.png)
+![The Call Stack pane](./reference-images/call-stack.png)
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:275]: add red box on blue arrow

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 12]
```diff
@@ -223,33 +248,31 @@ While paused on a line of code, use the **Scope** pane to view and edit the valu
...
 
 
 <!-- ====================================================================== -->
 ## View the call stack
 
 While paused on a line of code, use the **Call Stack** pane to view the call stack that got you to this point.
 
-<!--If you are working with async code, check the **Async** checkbox to enable async call stacks.  -->
-
 Click an entry to jump to the line of code where that function was called.  The blue arrow icon represents which function DevTools is currently highlighting.
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:273]:
```suggestion
Click an entry to jump to the line of code where that function was called.  The blue arrow icon represents which function DevTools is currently highlighting:
```
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/javascript/reference?branch=pr-en-us-2894#view-the-call-stack

File: microsoft-edge/devtools-guide-chromium/javascript/reference-images/scope.png
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference-images/scope.png:1]: add red box eg on Scope tab

https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/javascript/reference?branch=pr-en-us-2894#view-and-edit-properties-and-variables

**Review by mikehoffms — CHANGES_REQUESTED** [review]
Writer/Editor review

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md (hunk 5 above)
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/javascript/reference.md]: fixed
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/javascript/reference?branch=pr-en-us-2894#resume-script-execution

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md (hunk 11 above)
Comment (captainbrosset) [reply microsoft-edge/devtools-guide-chromium/javascript/reference.md:275]: Actually, I forgot to add a red box around the entire call stack pane. I'll do this.

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 13]
```diff
@@ -34,23 +34,31 @@ When fixing a bug, you often want to try out some changes to your JavaScript cod
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:34]: change 25 from:
This article covers how to use the debugger in Microsoft Edge DevTools, including how to set a line-of-code breakpoint. To set other types of breakpoints, see [Pause your code with breakpoints](breakpoints.md).
to:
This article covers how to use the debugger features in Microsoft Edge DevTools, including how to set a line-of-code breakpoint. To set other types of breakpoints, see [Pause your code with breakpoints](breakpoints.md).

Add "features", to give sharper contrast between the 3 articles discussed at top

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 14]
```diff
@@ -34,23 +34,31 @@ When fixing a bug, you often want to try out some changes to your JavaScript cod
 
 To view and edit a JavaScript file:
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:35]: pls try a rewrite of intro to streamline it, to focus on only the present article, and to clearly differentiate the 3 articles.

eg:
This article covers X aspects of Y.
See also:
Article 2 - covers Z aspects of Y.
Article 3 - covers W aspects of Y.

Probably delete my sentence "The tutorial has screen captures, so you can skim it." - more confusing than helpful.

"You can easily try out the debugger features by using the demo webpage." -- this is confusing: which demo for which article?  Probably need to name and specify which demo page, for which article.  Maybe delete that sentence, which isn't about the present article, and is confusing, excess detail, out-of-place.

File: microsoft-edge/devtools-guide-chromium/javascript/reference.md [hunk 15]
```diff
@@ -22,9 +22,11 @@ ms.date: 05/04/2021
    limitations under the License.  -->
 # JavaScript debugging features
 
-This article covers how to use the debugger in Microsoft Edge DevTools, including how to set a line-of-code breakpoint.  To set other types of breakpoints, see [Pause your code with breakpoints](breakpoints.md).
+This article covers how to use the debugger features in Microsoft Edge DevTools, including how to set a line-of-code breakpoint.
 
-To learn the basics of debugging, see [Get started debugging JavaScript](index.md), which is a tutorial that uses an existing, form-based webpage.  The tutorial has screen captures, so you can skim it.  You can easily try out the debugger features by using the demo webpage.
+See also:
+* [Pause your code with breakpoints](breakpoints.md).
+* [Get started debugging JavaScript](index.md).
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/javascript/reference.md:29]:
```suggestion
* [Get started debugging JavaScript](index.md) - a tutorial walkthrough.
* [Pause your code with breakpoints](breakpoints.md) - setting breakpoints in addition to line-of-code breakpoints.
```

Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/javascript/reference.md]: pushed commit

**Pull request merged**